zeus/device/cpu/intel.py

```diff
@@ -0,0 +1,151 @@
+"""Intel CPUs."""
+
+from __future__ import annotations
+
+import functools
+import os
+import contextlib
+from typing import Sequence, Dict, List
+
+import zeus.device.cpu.common as cpu_common
+from zeus.device.exception import ZeusBaseCPUError
+from zeus.utils.logging import get_logger
+
+logger = get_logger(name=__name__)
+
+DIR: str = "/sys/class/powercap/intel-rapl"
+
+def rapl_is_available() -> bool:
+    """Check if RAPL is available."""
+    if not os.path.exists(DIR):
+        logger.info("RAPL is not supported on this CPU.")
+        return False
+    logger.info("RAPL is available.")
+    return True
+
+class ZeusRAPLNotSupportedError(ZeusBaseCPUError):
+    """Zeus CPU exception class Wrapper for RAPL not supported on CPU."""
+
+    def __init__(self, message: str) -> None:
+        """Initialize Zeus Exception."""
+        super().__init__(message)
+
+class ZeusRAPLFileInitError(ZeusBaseCPUError):
+    """Zeus CPU exception class Wrapper for RAPL file initialization error on CPU."""
+
+    def __init__(self, message: str) -> None:
+        """Initialize Zeus Exception."""
+        super().__init__(message)
+
+class RAPLFile:
+    """RAPL File class for each RAPL file.
+
+    This class defines the interface for interacting with a RAPL file for a package. A package can
+    be a CPU or DRAM
+    """
+
+    def __init__(self, path: str) -> None:
+        self.path: str = path
+        self.energy_uj_path: str = os.path.join(path, "energy_uj")
+        try:
+            with open(os.path.join(path, "name"), 'r') as name_file:
+                self.name: str = name_file.read().strip()
+        except:
```

Comment:
You should always try to catch specific exceptions instead of doing a broad except.
See: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/broad-exception-caught.html

Not specifying any exception (`except:`) is equivalent to `except BaseException:`, which is a worse crime than `except Exception:`. It'll catch any exception, including system-exiting exceptions including `KeyboardInterrupt` or `SystemExit`. (See [Python exception hierarchy](https://docs.python.org/3/library/exceptions.html#exception-hierarchy))